src/js/serial.js

```diff
@@ -201,26 +220,34 @@ const serial = {
             for (let i = (self.onReceiveError.listeners.length - 1); i >= 0; i--) {
                 self.onReceiveError.removeListener(self.onReceiveError.listeners[i]);
             }
+            if (self.connectionId !== 'virtual') {
```

Comment:
@McGiverGim why is `!==` not allowed here?
Also spotted bug in line 224 below `==` should be `===`

Reply:
Weird, but I think it works fine.

Reply:
Should be `connectionType` instead of `connectionId`:

```suggestion
            if (self.connectionType !== 'virtual') {
```

Reply:
Please also solve the `sonarcloud` issues - it should be self explainable. I know out of experience, the editor doesn't remove trailing spaces itself. Don't know which linter solves this ;-)

Reply:
> @McGiverGim why is !== not allowed here? Sonar did not complain before about this❓
Also spotted bug in line 224 below == should be === (my fault)

This issue raises when the variables have different content. I have seen that we assign `true` or `false` to this variable in several places, so maybe assigning a `boolean` and a `string` makes Sonar detect it wrong (or it is too smart and has decided that the value is always `boolean` here and we don't have detected it.

> Should be connectionType instead of connectionId

It seems you have detected the bug, so it seems finally Sonar was smart. :)

Reply:
> Please also solve the sonarcloud issues - it should be self explainable. I know out of experience, the editor doesn't remove trailing spaces itself. Don't know which linter solves this ;-)

True, major part are very simple. You don't need to fix all of them, we usually admit the ones about refactor to reduce complexity, specially if the code is not new, but is very appreciated any refactor that reduces this.

You have all of the issues here: https://sonarcloud.io/project/issues?id=betaflight_betaflight-configurator&pullRequest=2283&resolved=false